clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -3783,10 +3823,20 @@ void TokenAnnotator::annotate(AnnotatedLine &Line) {
 static bool isFunctionDeclarationName(const LangOptions &LangOpts,
                                       const FormatToken &Current,
                                       const AnnotatedLine &Line,
+                                      const FormatStyle &Style,
                                       FormatToken *&ClosingParen) {
   if (Current.is(TT_FunctionDeclarationName))
     return true;
 
+  if (Current.is(TT_FunctionLikeOrFreestandingMacro) &&
+      std::find_if(
+          Style.KeywordedFunctionLikeMacros.begin(),
+          Style.KeywordedFunctionLikeMacros.end(),
+          [&Current](const FormatStyle::KeywordedFunctionLikeMacro &Decl) {
+            return Current.TokenText == Decl.Name;
+          }) != Style.KeywordedFunctionLikeMacros.end()) {
+    return true;
+  }
```

Comment:
It doesn't seem to make sense to annotate `Q_Property` as `TT_FunctionDeclarationName`. Why is this necessary?